internal/config/merge.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 Adevinta
+
+package config
+
+import (
+	"fmt"
+
+	"dario.cat/mergo"
+)
+
+// Merger has the ability to merge two configurations.
+type Merger interface {
+	Merge(dest, src Config) (Config, error)
+}
+
+// LavaMerger represents a merger for the Lava configuration.
+type LavaMerger struct{}
+
+// Merge merges two configurations. The values of the configuration
+// passed as first parameter will override those in the configuration
+// passed as second parameter.
+func (lm LavaMerger) Merge(dest, src Config) (Config, error) {
+	merged := Config{}
+	mergeOpts := []func(*mergo.Config){
+		mergo.WithOverride,
+		mergo.WithoutDereference,
+		mergo.WithAppendSlice,
+	}
+	if err := mergo.Merge(&merged, src, mergeOpts...); err != nil {
+		return Config{}, fmt.Errorf("merging src config into new config: %w", err)
+	}
+	if err := mergo.Merge(&merged, dest, mergeOpts...); err != nil {
+		return Config{}, fmt.Errorf("merging dest config into new config: %w", err)
+	}
```

Comment:
Aren't they inverted?

```suggestion
	if err := mergo.Merge(&merged, dest, mergeOpts...); err != nil {
		return Config{}, fmt.Errorf("merging dest config into new config: %w", err)
	}
	if err := mergo.Merge(&merged, src, mergeOpts...); err != nil {
		return Config{}, fmt.Errorf("merging src config into new config: %w", err)
	}
```

That would  explain my last question regarding the tests. I would expect `dest` to be overridden by `src`. That's actually the behavior of `mergo.Merge`. After looking more closely, I see that the doc comment explains the current behavior. But still.